### PR TITLE
Update sleep-interval to allow rand range in addition to int value #9930

### PR DESCRIPTION
**Pull request by scopeviable**
Adds random number feature to sleep_interval as suggested in #9930

### CODE REVIEW

**Comment by dstftw** [comment]
It should not change current behavior.

**Comment by shan3k** [comment]
ack, maybe I can update with --random_interval flag?

**Comment by expenses** [comment]
IMO I think it should work as default for `--sleep_interval NUM`, and randomly for `--sleep_internal NUM-NUM`.

**Comment by dstftw** [comment]
`--sleep-interval` supporting both variants and documented should be enough.

**Comment by shan3k** [comment]
@dstftw @expenses ack, let me have a go at that so. Still learning :+1:

File: youtube_dl/downloader/common.py [hunk 1]
```diff
@@ -343,6 +344,12 @@ def download(self, filename, info_dict):
             return True
 
         sleep_interval = self.params.get('sleep_interval')
+        if '-' in sleep_interval:
+            try:
+                start, stop = sleep_interval.split('-')
+                sleep_interval = random.randrange(int(start), int(stop))
```
Comment (dstftw) [inline youtube_dl/downloader/common.py]: floating point must be supported.

**Comment by dstftw** [comment]
Well, `start-end` format seems to be a not so good idea. Floating point numbers can be written in scientific notation, e.g. `3e-1`.

**Comment by expenses** [comment]
@dstftw Agreed, perhaps you could have `--sleep_interval time` for a constant time value, and `--sleep_interval start end` for a random time value?

**Comment by shan3k** [comment]
So, I guess in that case if we use `--sleep_interval time`` --sleep_interval start end` with support for floats, we should take input as a list and then parse?

**Comment by expenses** [comment]
@shan3k Seems to be the way to go.

**Pull request closed without merging**